org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/ContentModelPlugin.java

```diff
@@ -56,17 +56,17 @@ public void doSave(ISaveContext context) {
 			// The save is done for a given XML file
 			String documentURI = context.getUri();
 			DOMDocument document = context.getDocument(documentURI);
-			if (DOMUtils.isCatalog(document)) {
+			if (document != null && DOMUtils.isCatalog(document)) {
 				// the XML document which has changed is a XML catalog.
 				// 1) refresh catalogs
 				contentModelManager.refreshCatalogs();
-				// 2) Validate all opened XML files except the catalog which have changed
-				context.collectDocumentToValidate(d -> {
-					DOMDocument xml = context.getDocument(d.getDocumentURI());
-					xml.resetGrammar();
-					return !documentURI.equals(d.getDocumentURI());
-				});
 			}
+			// 2) Validate all opened XML files except the catalog which have changed
+			context.collectDocumentToValidate(d -> {
+				DOMDocument xml = context.getDocument(d.getDocumentURI());

```

Comment:
should you check `xml` for null?

Reply:
This case could occur when DOM document is parsing, but never seen this case.